### PR TITLE
Force systems with kernel option "quiet" to display TTY password prompt

### PR DESCRIPTION
**Pull request by ghfields**
### Motivation and Context

On systems that utilize TTY for password entry, if the kernel option "quiet"
is set, the system would appear to freeze on a blank screen, when in fact it is waiting for
password entry from the user.  

Since TTY is the fallback method, this has no effect on systemd or plymouth password prompting

### Description

By temporarily setting "printk" to "7", running the command, then
resuming with the original "printk" state, the user can see the password prompt.

### How Has This Been Tested?

Prompt now successfully displays on my system (Ubuntu 20.04 Daily) with "quiet" on kernel line

### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [X] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [X] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [X] All new and existing tests passed.
- [X] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by rlaager** [comment]
@ghfields Can you retry with the changes from https://github.com/zfsonlinux/zfs/pull/9723 and see if that fixes it?

**Comment by ghfields** [comment]
> @ghfields Can you retry with the changes from #9723 and see if that fixes it?

The #9723 fixes allow certain systems (Ubuntu) to avoid needing to go to TTY prompting.  This PR, however fixes systems that do utilize TTY and have "quiet" has the kernel option set.  With #9723, there are fewer systems, but doesn't eliminate all TTY use cases.

File: contrib/initramfs/scripts/zfs.in [hunk 1]
```diff
@@ -436,7 +436,11 @@ decrypt_fs()
 
 			# Prompt with ZFS tty, otherwise
 			else
+				# Setting "printk" temporarily to "7" will allow prompt even if kernel option "quiet"
+				storeprintk="$(cat /proc/sys/kernel/printk | awk '{print $1}')"
```
Comment (rlaager) [inline contrib/initramfs/scripts/zfs.in]: The `cat` is unnecessary if you just use: `awk '{print $1}' /proc/sys/kernel/printk`

File: contrib/initramfs/scripts/zfs.in [hunk 2]
```diff
@@ -410,11 +410,11 @@ decrypt_fs()
 	if [ "$(zpool list -H -o feature@encryption $(echo "${fs}" | awk -F\/ '{print $1}'))" = 'active' ]; then
 
 		# Determine dataset that holds key for root dataset
-		ENCRYPTIONROOT=$(${ZFS} get -H -o value encryptionroot "${fs}")
+		ENCRYPTIONROOT="$(get_fs_value "${fs}" encryptionroot)"
 
 		# If root dataset is encrypted...
 		if ! [ "${ENCRYPTIONROOT}" = "-" ]; then
-			KEYSTATUS="$(${ZFS} get -H -o value keystatus "${ENCRYPTIONROOT}")"
+			KEYSTATUS="$(get_fs_value "${ENCRYPTIONROOT}" keystatus)"
```
Comment (rlaager) [inline contrib/initramfs/scripts/zfs.in]: Personally, I'd prefer these be in separate commits (one PR seems fine), since they're unrelated.

**Comment by rlaager** [comment]
Other than eliminating the unnecessary `cat`, this looks good to me.

File: contrib/initramfs/scripts/zfs.in (hunk 1 above)
Comment (ghfields) [reply contrib/initramfs/scripts/zfs.in]: Thanks

File: contrib/initramfs/scripts/zfs.in (hunk 2 above)
Comment (ghfields) [reply contrib/initramfs/scripts/zfs.in]: Yea, was back and forth about if they were trivial enough with myself.  Additional PR inbound shortly.

Comment (ghfields) [reply contrib/initramfs/scripts/zfs.in]: Done.  Became #9736
